### PR TITLE
Completed pytests and test conversions tests for tests/test_test_list.py

### PR DESCRIPTION
**Pull request by hbrown-ST**

### CODE REVIEW

File: test/tobs/test_list.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    out, _ = capsys.readouterr()
+    out_to_check = """{"U20L0U01T:U20L0U01T": {"ASN_ID_DSN": "U20L0U01T", "ATODFILE": "DBU1405FU.R1H", "ATODGAIN": \
+"15.0", "BIASFILE": "E4P16298U.R2H", "DARKFILE": "E1Q14338U.R3H", "DGEOFILE": "SA119437U_DXY.FITS", "EXPSTART": \
+"1993-12-07 10:07:16.930000", "FILTER1": "0.0", "FILTER2": "0.0", "FILTNAM1": "UNDEFINED", "FILTNAM2": "UNDEFINED", \
+"FLATFILE": "N/A", "IDCTAB": "SAD1946EU_IDC.FITS", "IMAGETYP": "BIAS", "LRFWAVE": "0.0", "MASKFILE": "F8213081U.R0H", \
+"MEMBER_NAME_DSN": "U20L0U01T", "MEMBER_TYPE": "UNDEFINED", "MODE": "FULL", "OFFTAB": "S9M1329LU_OFF.FITS", "SERIALS": \
+"OFF", "SHADFILE": "E371355EU.R5H", "SHUTTER": "A", "WF4TFILE": "N/A", "INSTRUME": "WFPC2", "DATE-OBS": "1993-12-07", \
+"TIME-OBS": "10:07:16.929999"}}"""
+    assert out_to_check in out
+
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply test/tobs/test_list.py:236]: check in `capsys.err`. I see in line 227, that the error stream is ignored. Maybe not ignore it and use it. Note, read all about this at https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html

[the remark this answers is not in the document]
Comment (hbrown-ST) [reply test/tobs/test_list.py:236]: I have tried capturing the error as well, using out, err = capsys.readouterr() and the "err" is blank.

[the remark this answers is not in the document]
Comment (alphasentaurii) [reply test/tobs/test_list.py:236]: I've had more success using caplog. You need to make sure that the CRDS logger is being propagated to the pytest logger, e.g. `log.THE_LOGGER.logger.propagate=True`.  See examples here: https://github.com/spacetelescope/crds/blob/pytest-migration/test/core/test_rmap.py

[the remark this answers is not in the document]
Comment (hbrown-ST) [reply test/tobs/test_list.py:236]: Thanks @alphasentaurii , so caplog definitely does work, though I'm still getting the the log streamed to stout. I'd prefer it if only the pytest logging appeared.

[the remark this answers is not in the document]
Comment (hbrown-ST) [reply test/tobs/test_list.py:236]: It ends up looking like:


============================= test session starts ==============================
collecting ... collected 21 items

test_list.py::test_list_hst_mappings 
test_list.py::test_list_jwst_mappings 
test_list.py::test_list_cached_mappings 
test_list.py::test_list_references 
test_list.py::test_list_cached_references 
test_list.py::test_list_dataset_ids 
test_list.py::test_list_dataset_headers 
test_list.py::test_list_dataset_headers_json 
test_list.py::test_list_dataset_headers_bogus 
test_list.py::test_list_dataset_headers_id_expansions_only 
test_list.py::test_list_required_parkeys_pmap 
test_list.py::test_list_required_parkeys_imap 
test_list.py::test_list_required_parkeys_rmap 
test_list.py::test_list_resolve_contexts_range 
test_list.py::test_list_resolve_contexts_date 
test_list.py::test_list_remote_context 
test_list.py::test_list_operational_context 
test_list.py::test_list_references_by_context 
test_list.py::test_list_cat_mappings 
test_list.py::test_list_status 
test_list.py::test_list_config 

================== 21 passed, 8 warnings in 88.63s (0:01:28) ===================
PASSED                              [  4%]PASSED                             [  9%]PASSED                           [ 14%]PASSED                                [ 19%]PASSED                         [ 23%]PASSED                               [ 28%]PASSED                           [ 33%]PASSED                      [ 38%]PASSED                     [ 42%]CRDS - ERROR -  Failed fetching dataset parameters with repect to 'hst.pmap' for ['BAR:BAR'] : CRDS jsonrpc failure 'get_dataset_headers_by_id' OtherError: Can't determine instrument for dataset 'BAR'
PASSED        [ 47%]...


I don't like it but maybe it's fine.

**Pull request merged**